### PR TITLE
mimxrt: Bug fixes, small changes and additions.

### PR DESCRIPTION
**Pull request by robert-hh**
Some small changes and bug fixes.

- Change the RAM Pin configuration and fix the comment for the MIMXRT1176 port.
- Clear pending IRQ requests when Pin IRQ is enabled or disabled. Otherwise the IRQ handler may be called prematurely.

### CODE REVIEW

File: docs/mimxrt/quickref.rst [hunk 1]
```diff
@@ -438,6 +438,10 @@ The i.MXRT MCU supports battery backup of the RTC.  By connecting a battery of
 current drawn from the battery is ~20µA, which is rather high.  A CR2032 coin
 cell will last for about one year.
 
+Note: In v1.23.0 the support for subseconds was removed. When reading the RTC, 0 will
```
Comment (dpgeorge) [inline docs/mimxrt/quickref.rst:441]: Can you please link to the commit which removed subseconds?

Is it that the hardware just doesn't support subseconds?  But then how was it implemented before 1.23.0?

**Comment by robert-hh** [comment]
The hardware does not support subseconds. In 1.23.0 and earlier a subsecond mechanism was simulated using ticks_us(). But since RTC and ticks_us() use different clocks, that would drift away.
The change was done in commit 913f9ad5ad9bb446c6e4310b6b5eda310098a1cf

**Comment by dpgeorge** [comment]
OK, thanks for the info about subseconds.

Now merged.

**Comment by robert-hh** [comment]
Thank you.

**Pull request merged**